packages/api-server/src/base/gw-config.ts

```diff
@@ -0,0 +1,394 @@
+import { Uint32 } from "./types/uint";
+import { snakeToCamel } from "../util";
+
+export class GwConfig {
+  rpc: GodwokenClient;
+  private iNodeInfo: NodeInfo | undefined;
+  private iWeb3ChainId: HexNumber | undefined;
+  private iAccounts: ConfigAccounts | undefined;
+  private iEoaScripts: ConfigEoaScripts | undefined;
+  private iBackends: ConfigBackends | undefined;
+  private iGwScripts: ConfigGwScripts | undefined;
+  private iRollupConfig: RollupConfig | undefined;
+  private iRollupCell: RollupCell | undefined;
+  private iNodeMode: NodeMode | undefined;
+  private iNodeVersion: string | undefined;
+
+  constructor(rpcOrUrl: GodwokenClient | string) {
+    if (typeof rpcOrUrl === "string") {
+      this.rpc = new GodwokenClient(rpcOrUrl);
+      return;
+    }
+
+    this.rpc = rpcOrUrl;
+  }
+
+  async init(): Promise<GwConfig> {
+    this.iNodeInfo = await this.getNodeInfoFromRpc();
+
+    const ethAddrReg = await this.fetchEthAddrRegAccount();
+    const creator = await this.fetchCreatorAccount();
+    const defaultFrom = await this.fetchDefaultFromAccount();
+
+    this.iAccounts = {
+      polyjuiceCreator: creator,
+      ethAddrReg,
+      defaultFrom,
+    };
+
+    this.iEoaScripts = toConfigEoaScripts(this.nodeInfo);
+    this.iGwScripts = toConfigGwScripts(this.nodeInfo);
+    this.iBackends = toConfigBackends(this.nodeInfo);
+    this.iWeb3ChainId = this.nodeInfo.rollupConfig.chainId;
+    this.iRollupCell = this.nodeInfo.rollupCell;
+    this.iRollupConfig = this.nodeInfo.rollupConfig;
+    this.iNodeMode = this.nodeInfo.mode;
+    this.iNodeVersion = this.nodeInfo.version;
+
+    return this;
+  }
+
+  public get web3ChainId(): HexNumber {
+    return this.iWeb3ChainId!;
+  }
+
+  public get accounts(): ConfigAccounts {
+    return this.iAccounts!;
+  }
+
+  public get backends(): ConfigBackends {
+    return this.iBackends!;
+  }
+
+  public get eoaScripts(): ConfigEoaScripts {
+    return this.iEoaScripts!;
+  }
+
+  public get gwScripts(): ConfigGwScripts {
+    return this.iGwScripts!;
+  }
+
+  public get rollupConfig(): RollupConfig {
+    return this.iRollupConfig!;
+  }
+
+  public get rollupCell(): RollupCell {
+    return this.iRollupCell!;
+  }
+
+  public get nodeMode(): NodeMode {
+    return this.iNodeMode!;
+  }
+
+  public get nodeVersion(): string {
+    return this.iNodeVersion!;
+  }
+
+  private get nodeInfo(): NodeInfo {
+    return this.iNodeInfo!;
+  }
+
+  private async getNodeInfoFromRpc() {
+    const nodeInfo = await this.rpc.getNodeInfo();
+    return toApiNodeInfo(nodeInfo);
+  }
+
+  private async fetchCreatorAccount(ethAddrRegId?: HexNumber) {
+    const ckbSudtId = new Uint32(+CKB_SUDT_ID).toLittleEndian();
+    ethAddrRegId ||= new Uint32(
+      +(await this.fetchEthAddrRegAccount()).id
+    ).toLittleEndian();
+
+    const creatorScriptArgs =
+      this.nodeInfo.rollupCell.typeHash +
+      ckbSudtId.slice(2) +
+      ethAddrRegId.slice(2);
```

Comment:
Please update the `creatorScriptArgs` according to this PR:
- https://github.com/nervosnetwork/godwoken/pull/661/files#diff-7e54fb295e55eacd2f6758ae86464d0d837d065bea52948cf3fd8f2bb3b8f5c1L59

Reply:
updated